src/Http/Routing/src/Matching/MatcherPolicy.cs

```diff
@@ -24,5 +27,30 @@ public abstract class MatcherPolicy
         /// property.
         /// </summary>
         public abstract int Order { get; }
+
+        /// <summary>
+        /// Returns a value that indicates whether the provided <paramref name="endpoints"/> contains
+        /// one or more dynamic endpoints.
+        /// </summary>
+        /// <param name="endpoints">The set of endpoints.</param>
+        /// <returns><c>true</c> if a dynamic endpoint is found; otherwise returns <c>false</c>.</returns>
+        protected static bool ContainsDynamicEndpoints(IReadOnlyList<Endpoint> endpoints)
```

Comment:
Who would use this? It looks like it is only called from tests at the moment.

Reply:
A matcher policy that wants to know if there's any dynamic behaviour inside its `AppliesToEndpoints` method. Since you and @pranavkm are both stuck on this I'll use it just to make y'all happy.